src/java.base/share/classes/java/lang/ModuleLayer.java

```diff
@@ -147,9 +149,17 @@
 
 public final class ModuleLayer {
 
-    // the empty layer
-    private static final ModuleLayer EMPTY_LAYER
-        = new ModuleLayer(Configuration.empty(), List.of(), null);
+    // the empty layer (may be initialized from the CDS archive)
+    private static @Stable ModuleLayer EMPTY_LAYER;
+
+    static {
+        // Initialize EMPTY_LAYER from the archive.
```

Comment:
Style wide you can drop L154 and L156 and the comment on EMPTY_LAYER will cover its initialization too.

Otherwise looks okay but the more archiving we add does make it harder to change this code.

Reply:
Fixed according to your comments. Thanks! 

I was surprised to find CDS specific code in the java layer as well, but it's not the design choice I believe I can make with this bug fix.